### DISCUSSION TITLE
MsgHdrProtoBuf deserialization results in excessive memory usage and inability to release memory in a timely manner

### DISCUSSION THREAD

**Discussion by yingpanwang** [SteamRE/SteamKit#1363] [General]
I found that memory usage keeps increasing when using dotMemory, and many references come from MsgHdrProtoBuf.
![90c19190-411f-44ab-b2f2-b6cf0c9e4c5e](https://github.com/SteamRE/SteamKit/assets/49632937/01696158-6c31-4b86-986d-c90cc41afe02)

**Comment by yaakov-h** [comment]
What version of SteamKit and runtime are you on?

**Reply by yingpanwang** [reply]
2.5.0-beta 1

**Reply by yingpanwang** [reply]
Runtime  .Net 8
SteamKit 2.5.0-beta 1
